### ISSUE TITLE
set_rate_limit not properly rate limiting for my connection

### ISSUE THREAD

**Issue by aristeia** [alastair/python-musicbrainzngs#198]
Hi.
I'm encountering an error when trying to rate limit my application which uses musicbrainzngs. I've tried invoking calls to mb.set_rate_limit() with either no parameters (thus getting default values) or very high values (e.g. mb.set_rate_limit(15.0)), but still run into overusage problems. Although this throttling is being done on an IP level (based on my Source IP address), I'm almost certain that no one else on my IP is using musicbrainz frequently. Could there be an issue with how the package is limiting requests?

I'm running 0.7.dev0

**Comment by alastair** [comment]
When you say that you're running into overusage problems do you mean that you keep getting errors when you query musicbrainz?
Unfortunately, musicbrainz is having some pretty serious growing pains, and the only solution is to move to a new hosting provider which will give them more bandwidth. Currently there are some quite strict rate limits in place, which are currently global (as opposed to per-ip). There is some discussion about this here: https://community.metabrainz.org/t/state-of-things-q2-2016/38918/3
In short, unfortunately there's not much you can do until this change is done, unless you want to host your own musicbrainz mirror.

**Comment by aristeia** [comment]
That makes sense; thanks for the update.

**Comment by mineo** [comment]
It looks like this isn't a bug in pymngs, so I'll close the ticket. If I misunderstood it, feel free to leave a comment and I'll reopen the ticket.
